### ISSUE TITLE
Rust 1.18 regression, rustasm6502 0.1.2, no rules expected token 'main'

### ISSUE THREAD

**Issue by brson** [rust-lang/rust#41803]
https://github.com/jonas-schievink/rustasm6502

```
brian@ip-10-145-43-250:~/dev/rustasm6502⟫ git log -1
commit e2e2d558b251e728232273f8019f5a37eb2e112e
Author: Jonas Schievink <jonasschievink@gmail.com>
Date:   Wed Aug 10 23:04:27 2016 +0200

    Update my email
```

```
brian@ip-10-145-43-250:/mnt2/dev⟫ rustc +beta -Vv
rustc 1.18.0-beta.1 (4dce67253 2017-04-25)
binary: rustc
commit-hash: 4dce67253d67785c74c8d270c38d774edd0deed8
commit-date: 2017-04-25
host: x86_64-unknown-linux-gnu
release: 1.18.0-beta.1
LLVM version: 3.9

```

```
brian@ip-10-145-43-250:~/dev/rustasm6502⟫ cargo +beta test
   Compiling rustasm6502 v0.1.1 (file:///mnt2/dev/rustasm6502)
error: no rules expected the token `main`
    --> src/lib.rs:1083:24
     |
1083 |     assert_eq!(my_map!(main), 0);
     |                --------^^^^- in this macro invocation

error: Could not compile `rustasm6502`.
Build failed, waiting for other jobs to finish...
error: build failed
```

cc @jonas-schievink @jseyfried

**Comment by jonas-schievink** [comment]
I've turned on nightly Travis builds so I get notified when something breaks. This crate seems like a good candidate for finding regressions in macros earlier.

**Comment by Mark-Simulacrum** [comment]
Introduced by fe151194e9a919ab23c2bb356767a5cbfa3811e7 -- https://github.com/rust-lang/rust/pull/40597, cc @jseyfried

**Comment by nikomatsakis** [comment]
@jseyfried -- any progress? should somebody else take a look?

**Comment by jseyfried** [comment]
Fixed in #42005.
